src/libtomahawk/Query.cpp

```diff
@@ -403,12 +403,12 @@ Query::resultSorter( const result_ptr& left, const result_ptr& right )
         {
             return false;
         }
-        if ( !right->isPreview() )
+        if ( left->isPreview() != right->isPreview() )
         {
-            return false;
+            return !left->isPreview();
         }
 
-        return true;
+        return right->resolvedBy()->weight() < left->resolvedBy()->weight();
```

Comment:
Can you make it "left > right" - nicer to the eye, seeing how all the other comparisons in this method stick to left being on the, er, left side.